src/mender-update/daemon/context.cpp

```diff
@@ -351,6 +351,68 @@ error::Error Context::SaveDeploymentStateData(StateData &state_data) {
 		return SaveDeploymentStateData(txn, state_data);
 	});
 }
+static error::Error UnmarshalJsonStateDataVersion1(const json::Json &json, StateData &state_data) {
+#define SetOrReturnIfError(dst, expr) \
+	if (!expr) {                      \
+		return expr.error();          \
+	}                                 \
+	dst = expr.value()
+
+#define DefaultOrSetOrReturnIfError(dst, expr, def)                            \
+	if (!expr) {                                                               \
+		if (expr.error().code == kv_db::MakeError(kv_db::KeyError, "").code) { \
+			dst = def;                                                         \
+		} else {                                                               \
+			return expr.error();                                               \
+		}                                                                      \
+	} else {                                                                   \
+		dst = expr.value();                                                    \
+	}
+
+	auto exp_int = json.Get("Version").and_then(json::ToInt);
+	SetOrReturnIfError(state_data.version, exp_int);
+
+	if (state_data.version != 1) {
+		return error::Error(
+			make_error_condition(errc::not_supported),
+			"State Data version not supported by this client (" + to_string(state_data.version)
```

Comment:
I think the error here is misleading. I would say that calling this function with version != 1 is a programming error.

Reply:
Agreed - fixing